File: backend/entity/AcademicStaff.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type AcademicStaff struct {
	gorm.Model
	AcademicPosition 	string 		`json:"academic_position"`
	FirstName 			string    	`json:"first_name"`
	LastName  			string    	`json:"last_name"`
	Birthday  			time.Time 	`json:"birthday"`
	Age              	int    		`json:"age"`
	UniversityID     uint       `json:"university_id"`
	University       University `gorm:"foreignKey:UniversityID"`

	FacultyID        uint     `json:"faculty_id"`
	Faculty          Faculty  `gorm:"foreignKey:FacultyID"`

	ProgramID        uint     `json:"program_id"`
	Program          Program  `gorm:"foreignKey:ProgramID"`

	UserID    uint  `json:"user_id"`
	User      User   `gorm:"foreignKey:UserID"`
	
	AddressID uint    `json:"address_id"`
	Address   Address `gorm:"foreignKey:AddressID"`
	
	GenderID  uint   `json:"gender_id"`
	Gender    Gender `gorm:"foreignKey:GenderID"`

	ContactID  uint   `json:"contact_id"`
	Contact    Contact `gorm:"foreignKey:ContactID"`

}

File: frontend/src/interfaces/AcademicStaff.ts
import type { UserInterface } from "./User";
import type { AddressInterface } from "./Address";
import type { AdminInterface } from "./Admin";
import type { GenderInterface } from "./Gender";
import type { UniversityInterface } from "./UniversityInterface";
import type { FacultyInterface } from "./FacultyInterface";
import type { ProgramInterface } from "./ProgramInterface";
import type { ContactInterface } from "./Contact";

export interface AcademicStaffInterface {

  id: number;
  ID?: number;
  academic_position: string;
  age: number;
  university_id?: number;
  University: UniversityInterface;

  faculty_id?: number;
  Faculty: FacultyInterface;

  program_id?: number;
  Program: ProgramInterface;

  verify: boolean;


  UserID?: number;
  address_id?: number;
  AdminID?: number;
  gender_id?: number;

  first_name?: string;
  last_name?: string;

  User?: UserInterface;
  Address?: AddressInterface;
  Admin?: AdminInterface;
  Gender?: GenderInterface;
  Contact?: ContactInterface;
}
